### DISCUSSION TITLE
Are yarn pnp and pnpm similar?

### DISCUSSION THREAD

**Discussion by bestseob93** [yarnpkg/berry#3662] [Q&A]
Hi, thanks for your works.

While comparing pnpm and pnp, a question arose.
`.pnp.cjs`'s `"packageRegistryData"` and `.pnpm`folder structure look similar.

is there any difference except hard/symlink and zip archive?

**Comment by arcanis** [comment]
There are a few other differences, the main one imo being that pnpm-style installs still use the filesystem, and as a result cannot perfectly install all dependency trees (cf [this comment](https://github.com/yarnpkg/berry/blob/e95a1df42f6b94fc7a308af1b038e43c7278ac5a/packages/plugin-pnpm/sources/PnpmLinker.ts#L265-L275)).

**Reply by bestseob93** [reply]
Thanks @arcanis  :)
so, Is the strategy to prevent duplicate instantiation the similar?
